src/ast/semantic_analyser.cpp

```diff
@@ -1725,6 +1793,34 @@ void SemanticAnalyser::visit(AttachPoint &ap)
       }
     }
   }
+  else if (ap.provider == "kfunc" || ap.provider == "kretfunc")
+  {
+#ifdef HAVE_KFUNC
+    bool supported = bpftrace_.btf_.has_data();
+#else
+    bool supported = false;
+#endif
+
+    if (!supported)
+    {
+      error("kfunc/kretfunc not available for your kernel version.", ap.loc);
+      return;
+    }
+
+    auto ap_map = bpftrace_.btf_ap_args_;
```

Comment:
I think we can avoid a copy here
```suggestion
    const auto& ap_map = bpftrace_.btf_ap_args_;
```

Reply:
ok